ecms_base/themes/custom/ecms/templates/paragraphs/paragraph--icon-button.html.twig

```diff
@@ -0,0 +1,80 @@
+{#
+/**
+ * @file
+ * Default theme implementation to display an icon button paragraph.
+ *
+ * Available variables:
+ * - paragraph: Full paragraph entity.
+ *   Only method names starting with "get", "has", or "is" and a few common
+ *   methods such as "id", "label", and "bundle" are available. For example:
+ *   - paragraph.getCreatedTime() will return the paragraph creation timestamp.
+ *   - paragraph.id(): The paragraph ID.
+ *   - paragraph.bundle(): The type of the paragraph, for example, "image" or "text".
+ *   - paragraph.getOwnerId(): The user ID of the paragraph author.
+ *   See Drupal\paragraphs\Entity\Paragraph for a full list of public properties
+ *   and methods for the paragraph object.
+ * - content: All paragraph items. Use {{ content }} to print them all,
+ *   or print a subset such as {{ content.field_example }}. Use
+ *   {{ content|without('field_example') }} to temporarily suppress the printing
+ *   of a given child element.
+ * - attributes: HTML attributes for the containing element.
+ *   The attributes.class element may contain one or more of the following
+ *   classes:
+ *   - paragraphs: The current template type (also known as a "theming hook").
+ *   - paragraphs--type-[type]: The current paragraphs type. For example, if the paragraph is an
+ *     "Image" it would result in "paragraphs--type--image". Note that the machine
+ *     name will often be in a short form of the human readable label.
+ *   - paragraphs--view-mode--[view_mode]: The View Mode of the paragraph; for example, a
+ *     preview would result in: "paragraphs--view-mode--preview", and
+ *     default: "paragraphs--view-mode--default".
+ * - view_mode: View mode; for example, "preview" or "full".
+ * - logged_in: Flag for authenticated user status. Will be true when the
+ *   current user is a logged-in member.
+ * - is_admin: Flag for admin user status. Will be true when the current user
+ *   is an administrator.
+ *
+ * @see template_preprocess_paragraph()
+ *
+ * @ingroup themeable
+ */
+#}
+
+{# Determine which icon to render #}
+{% set media_library_icon = FALSE %}
+
+{% if content.field_icon_library_icon.0.media_library_icon %}
+  {% set media_library_icon = TRUE %}
+  {% set icon = content.field_icon_library_icon.0.media_library_icon %}
+{% else %}
+  {% set icon = content.field_icon_library_icon.0.pl_icon %}
+{% endif %}
+
+{# Link variables #}
+{% if not paragraph.field_link is empty %}
+  {% set link_title = content.field_link.0['#title'] %}
+
+  {# Get link url #}
+  {% if paragraph.field_link.0.url.external %}
+    {% set link_url = content.field_link.0['#url'].getUri() %}
+  {% else %}
+    {% set link_url = content.field_link.0['#url'] %}
```

Comment:
Will this work if the link has been translated? Might need the langcode test here

Reply:
Yes, since it is using the content object rather than the node object it should work.